utils/hostapi.v2.d.ts

```diff
@@ -0,0 +1,335 @@
+
+    /**
+     * The tags associated with this resource.
+     */
+    readonly tags?: {
+        [propertyName: string]: string;
+    };
+
+    /**
+     * A copy of the raw resource.
+     */
+    readonly raw: {};
+}
+
+export interface ViewPropertiesModel {
+    /**
+     * File name displayed in VS Code.
+     */
+    label: string;
+
+    /**
+     * Raw data associated with the resource to populate the properties file.
+     */
+    data: {};
+}
+
+/**
+ * Represents a model of an individual Azure resource or its child items.
+ */
+export interface AzureResourceModel extends ResourceModelBase {
+    /**
+     * The Azure ID of this resource.
+     *
+     * @remarks This property is expected to be implemented on "application-level" resources, but may also be applicable to its child items.
+     */
+    readonly azureResourceId?: string;
+
+    /**
+     * The URL of the area of Azure portal related to this item.
+     */
+    readonly portalUrl?: vscode.Uri;
+
+    /**
+     * Define to enable the "View Properties" command.
+     */
+    readonly viewProperties?: ViewPropertiesModel;
+}
+
+/**
+ * A provider for supplying items for the Azure resource tree (e.g. Cosmos DB, Storage, etc.).
+ */
+export type AzureResourceProvider = ResourceProvider<AzureSubscription, AzureResource>;
+
+/**
+ * A provider for visualizing items in the Azure resource tree (e.g. Cosmos DB, Storage, etc.).
+ */
+export type AzureResourceBranchDataProvider<TModel extends AzureResourceModel> = BranchDataProvider<AzureResource, TModel>;
+
+/**
+ * Respresents a specific type of workspace resource.
+ *
+ * @remarks This value should be unique across all types of workspace resources.
+ */
+type WorkspaceResourceType = string;
+
+/**
+ * An indivdual root resource for a workspace.
+ */
+export interface WorkspaceResource extends ResourceBase {
+    /**
+     * The folder to which this resource belongs.
+     */
+    readonly folder: vscode.WorkspaceFolder;
+
+    /**
+     * The type of this resource.
+     *
+     * @remarks This value is used to map resources to their associated branch data provider.
+     */
+    readonly resourceType: WorkspaceResourceType;
+}
+
+/**
+ * Represents a model of an individual workspace resource or its child items.
+ */
+export type WorkspaceResourceModel = ResourceModelBase;
+
+/**
+ * A provider for supplying items for the workspace resource tree (e.g., storage emulator, function apps in workspace, etc.).
+ */
+export type WorkspaceResourceProvider = ResourceProvider<vscode.WorkspaceFolder, WorkspaceResource>;
+
+/**
+ * A provider for visualizing items in the workspace resource tree (e.g., storage emulator, function apps in workspace, etc.).
+ */
+export type WorkspaceResourceBranchDataProvider<TModel extends WorkspaceResourceModel> = BranchDataProvider<WorkspaceResource, TModel>;
+
+type ResourceGroupsTreeDataProvider = Pick<vscode.TreeDataProvider<unknown>, 'getChildren' | 'getTreeItem'>;
```

Comment:
Can you add a comment to explain why this is being added? (i.e. to scope down what is allowed to be used from the TDPs)